api/src/singleFunction/checkPushToken.js

```diff
@@ -0,0 +1,43 @@
+import { jwtDecode } from 'jwt-decode';
+
+/**
+ * Checks the pushToken can be used to push on a specified repository
+ */
+export async function checkPushToken({ pushToken, repository }) {
+  const decoded = jwtDecode(pushToken);
+
+  if (!Array.isArray(decoded?.access)) {
+    throw Error('Missing access in jwt');
+  }
+  // check claims
+  let hasPushAccess = false;
+  decoded?.access?.forEach((access) => {
+    if (
+      // repo
+      access?.type === 'repository' &&
+      access?.name === repository &&
+      // push action
+      Array.isArray(access?.actions) &&
+      access?.actions?.includes('pull') &&
+      access?.actions?.includes('push')
+    ) {
+      hasPushAccess = true;
+    }
+  });
```

Comment:
You can simplify this code using Array.prototype.some(), which is more concise and better suited for this use case since it stops as soon as a match is found. Here’s the improved version:
```suggestion
const hasPushAccess = decoded?.access?.some(
  (access) =>
    access?.type === 'repository' &&
    access?.name === repository &&
    Array.isArray(access?.actions) &&
    access?.actions.includes('pull') &&
    access?.actions.includes('push')
);
```

Reply:
great :+1: